src/Build.UnitTests/BackEnd/TaskExecutionHost_Tests.cs

```diff
@@ -1560,7 +1560,8 @@ private ProjectInstance CreateTestProject()
                 </Project>
                 ");
 
-            Project project = new Project(XmlReader.Create(new StringReader(projectFileContents)));
```

Comment:
This pattern is so prevalent, I wonder if we should make a new test helper to make it smoother, something like

```c#
public class ProjectFromString : IDisposable
{
    public Project Project {get;init;}
    
    private XmlReader _reader;
    
    public ProjectFromString(string s)
    {
        _reader = XmlReader.Create(new StringReader(s));
        
        Project = new(_reader);
    }

    public void Dispose()
    {
        ((IDisposable)_reader).Dispose();
    }
}
```

And replace the existing calls with

```c#
using var projectFromString = new (projectFileContents);
Project project = projectFromString.Project;
```

? I guess the impact at every callsite is about the same, though.

Reply:
Updated